modules/core/src/v2/coins/utxo/sign.ts

```diff
@@ -78,3 +90,87 @@ export function signWalletTransactionWithUnspent(
     unspent.value
   );
 }
+
+export class InputSigningError extends Error {
+  constructor(public inputIndex: number, public unspent: Unspent, public reason: Error | string) {
+    super(`singing error at input ${inputIndex}: unspentId=${unspent.id}: ${reason}`);
+  }
+}
+
+export class TransactionSigningError extends Error {
+  constructor(signErrors: InputSigningError[], verifyError: InputSigningError[]) {
+    super(
+      `sign errors at inputs: [${signErrors.join(',')}], ` +
+        `verify errors at inputs: [${verifyError.join(',')}], see log for details`
+    );
+  }
+}
+
+/**
+ * Sign all inputs of a wallet transaction and verify signatures after signing.
+ * Collects and logs signing errors and verification errors, throws error in the end if any of them
+ * failed.
+ *
+ * @param transaction - wallet transactions to be signed
+ * @param unspents - transaction unspents
+ * @param walletSigner - singing parameters
```

Comment:
"signing" although "singing parameters" has a nice ring to it.

Reply:
:musical_note:

Reply:
fixed